internal/models/email.go

```diff
@@ -8,11 +8,11 @@ import (
 	"github.com/lib/pq"
 )
 
-type EmailModel struct {
+type EmailModelPostgres struct {
 	DB *sql.DB
 }
```

Comment:
Code doesn't have to know what exact db we use. For example now we use PostgreSQL and want to migrate to MSSQL. Then we need to change every '-Postgres' ending.
Also 'Model' word in very confusing. It is repository, so I would name EmailRepository and tranfer it to repository package

Reply:
Which code doesn't have to know what exact DB we use? Indeed, high-level modules shouldn't know about details of low-level ones. But in this case, `EmailModelPostgres` is a **concrete** implementation of `EmailModel` interface, because of PostgreSQL specific error codes and special parameterized query syntax. 
Okay, we want to migrate to MSSQL, so what are we gonna do? Maybe just create a new implementation of `EmailModel`, something like `EmailModelMSSQL` and pass it to the service layer.

I'll change Model to Repository in the next commit. Agree!

Reply:
But now you don't have interface for PostgresEmailRepository, it is used directly in code. 

main.go:
`81	emailRepository := &repositories.PostgresEmailRepository{DB: db}`

Reply:
Interface for `PostgresEmailRepository` is `services.EmailRepository`. Yeah, I instantiate it directly, because i don't have DI container